pkg/function/backup_data.go

```diff
@@ -128,34 +130,36 @@ func (*backupDataFunc) RequiredArgs() []string {
 		BackupDataIncludePathArg, BackupDataBackupArtifactPrefixArg}
 }
 
-func backupData(ctx context.Context, cli kubernetes.Interface, namespace, pod, container, backupArtifactPrefix, includePath, encryptionKey string, tp param.TemplateParams) (string, string, error) {
+func backupData(ctx context.Context, cli kubernetes.Interface, namespace, pod, container, backupArtifactPrefix, includePath, encryptionKey string, tp param.TemplateParams) (string, string, string, string, error) {
 	pw, err := getPodWriter(cli, ctx, namespace, pod, container, tp.Profile)
 	if err != nil {
-		return "", "", err
+		return "", "", "", "", err
 	}
 	defer cleanUpCredsFile(ctx, pw, namespace, pod, container)
 	if err = restic.GetOrCreateRepository(cli, namespace, pod, container, backupArtifactPrefix, encryptionKey, tp.Profile); err != nil {
-		return "", "", err
+		return "", "", "", "", err
 	}
 
 	// Create backup and dump it on the object store
 	backupTag := rand.String(10)
 	cmd, err := restic.BackupCommandByTag(tp.Profile, backupArtifactPrefix, backupTag, includePath, encryptionKey)
 	if err != nil {
-		return "", "", err
+		return "", "", "", "", err
 	}
 	stdout, stderr, err := kube.Exec(cli, namespace, pod, container, cmd, nil)
 	format.Log(pod, container, stdout)
 	format.Log(pod, container, stderr)
 	if err != nil {
-		return "", "", errors.Wrapf(err, "Failed to create and upload backup")
+		return "", "", "", "", errors.Wrapf(err, "Failed to create and upload backup")
 	}
 	// Get the snapshot ID from log
 	backupID := restic.SnapshotIDFromBackupLog(stdout)
 	if backupID == "" {
-		return "", "", errors.New("Failed to parse the backup ID from logs")
+		return "", "", "", "", errors.New("Failed to parse the backup ID from logs")
 	}
-	return backupID, backupTag, nil
+	// Get the snapshot ID from log
+	fileCount, backupSize := restic.SnapshotStatsFromBackupLog(stdout)
```

Comment:
Note I am not doing any error checking here. If the parse failed I'm returning the values as-are (empty strings).

Reply:
Let's at least log a failure to parse at debug level.

Reply:
log.Debug added